### PR TITLE
Return 1 if error

### PR DESCRIPTION
**Pull request by tagomaru**
This addresses #118, but still WIP.

@rocky 
This is my concept. Could you give me any comments?
If this looks fine to you, i continue to work on this for testing and improving.

Thx!

### CODE REVIEW

**Comment by tagomaru** [comment]
CircleCI failed.
I guess we have to change [this code](https://github.com/ConsenSys/truffle-security/blob/master/.circleci/config.yml#L66).
After rocky review, i will chage it as well.

File: index.js [hunk 1]
```diff
@@ -20,5 +20,15 @@ module.exports = async (config) => {
     if (config.help) return helpers.printHelpMessage();
     if (config.version) return helpers.printVersion();
 
-    return await helpers.analyze(config);
+    try {
+        const returnCode = await helpers.analyze(config);
+        if (returnCode === 0 || returnCode === undefined) {
```
Comment (rocky) [inline index.js]: Thoughts on the pros and cons of `returnCode === undefined` being treated as no problems in contrast to there was some problem? 

If analyze somehow returns undefined, wouldn't we want to know that?

**Comment by tagomaru** [comment]
> This is good so far. As you say, it would be good to have some tests for the return code. It doesn't seem that hard to write a unit test for say doReport or ghettoReport where there are errors and not errors.

@rocky 
Thank you for your comment !
Yes, I will write the tests.

> As an aside, a longer term plan for that integration test is to encapsulate the testing code into a standalone script so that you don't have to check that kind of test only via CI, but can do a similar thing in the checked-out code. #85 was my lame attempt to start doing that.

Thx for your info.

File: index.js (hunk 1 above)
Comment (tagomaru) [reply index.js]: Good question.
I wrote it for possible non-returns which can be caused by my mistake.

However, as you said, we should handle this with logging and abnormal termination.
I will do that, thx!

**Comment by tagomaru** [comment]
This commit addresses the below things.

- Small change
- Write test for this PR
- change config.yml for avoiding failure of integration test. And set --no-progress option.

File: .circleci/config.yml [hunk 2]
```diff
@@ -63,7 +63,7 @@ jobs:
             truffle compile
             CONTRACT_NAME=$(pwgen 16)
             sed -i "s/\"contractName\": \"FunctionTypes\"/\"contractName\": \"${CONTRACT_NAME}\"/" build/contracts/FunctionTypes.json
-            truffle run verify --style json | tee result.json
+            echo $(truffle run verify --style json --no-progress) | tee result.json
```
[the remark this answers is not in the document]
Comment (rocky) [reply .circleci/config.yml]: I agree.

**Comment by b-mueller** [comment]
Bounty is paid.

**Pull request merged**